### ISSUE TITLE
Should `SomeStrEnum(str, Enum)` require `SLOT000`?

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#5748]
### Discussed in https://github.com/astral-sh/ruff/discussions/5747

<div type='discussions-op-text'>

<sup>Originally posted by **jamesbraza** July 13, 2023</sup>
Please see the below file with Python 3.10.10:

```python
from enum import Enum

class SomeStrEnum(str, Enum):
    KEY = "value"
```

Running `ruff==0.0.278` on this:

```bash
> ruff --isolated --select="SLOT" a.py
a.py:3:7: SLOT000 Subclasses of `str` should define `__slots__`
Found 1 error.
```

Running `flake8-slots==0.1.5` with `flake8==4.0.1` on this, there's no error:

```bash
> flake8 a.py
```

Fwiw, Python 3.11's built-in [`enum.StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum) doesn't seem to define `__slots__`.

I am not sure if there should or should not be `SLOT000` for this.  Looks like @qdegraaf added `SLOT` rule in `ruff==0.0.273`, so notifying him.

I am thinking either:
- This is a false positive of `SLOT000`
- Or should the built-in `StrEnum` define `__slots__`? </div>

**Comment by charliermarsh** [comment]
Looks like `flake8-slots` omits enums: https://github.com/python-formate/flake8-slots/blob/0b1ecb9076c6194f29265b7bfb904d63073bce08/flake8_slots/__init__.py#L102.

**Comment by qdegraaf** [comment]
Don't know how I missed that. My bad. thanks for the quick fix @charliermarsh !

**Comment by charliermarsh** [comment]
Yeah, we don't trace across base classes right now. You'd need to add `Enum` as a parent class to have it detected properly.

**Comment by sujuka99** [comment]
> Yeah, we don't trace across base classes right now. You'd need to add `Enum` as a parent class to have it detected properly.

Thanks for the quick answer :)
